packages/overlays/stories/index.stories.mdx

```diff
@@ -0,0 +1,646 @@
+  html`
+    <span>${appState.opened}</span>
+  `,
+);
+
+// Function that syncs our appState (and also Node reference to make it visible in our demo)
+function onOpenClosed(ev) {
+  appState.opened = ev.target.opened;
+  openedStateNode.innerText = appState.opened;
+}
+```
+
+```html
+<div>
+  appState.opened: ${openedStateNode}
+</div>
+<lion-demo-overlay .opened="${appState.opened}" @opened-changed=${onOpenClosed}>
+  <button slot="invoker">Overlay</button>
+  <div slot="content" class="demo-overlay">
+    Hello! You can close this notification here:
+    <button
+      class="close-button"
+      @click=${e => e.target.dispatchEvent(new Event('close-overlay', { bubbles: true }))}
+    >
+      ⨯
+    </button>
+  </div>
+</lion-demo-overlay>
+```
+
+## Intercepting open/close
+
+It is possible to intercept the open/close by making sure you fire `Event.preventDefault();` on the `before-close` or `before-open` events.
+
+<Story name="Intercept open/close">
+  {() => {
+    let shouldIntercept = true;
+    let shouldInterceptButton;
+    function toggleIntercept() {
+      shouldIntercept = !shouldIntercept;
+      shouldInterceptButton.textContent = shouldIntercept;
+    }
+    function intercept(ev) {
+      if (shouldIntercept) {
+        ev.preventDefault();
+      }
+    }
+    shouldInterceptButton = renderLitAsNode(
+      html`
+        <button @click="${toggleIntercept}">${shouldIntercept}</button>
+      `,
+    );
+    return html`
+      <div>
+        Toggle shouldIntercept: ${shouldInterceptButton}
+      </div>
+      <lion-demo-overlay @before-closed=${intercept} @before-opened=${intercept}>
+        <button slot="invoker">Overlay</button>
+        <div slot="content" class="demo-overlay">
+          Hello! You can close this notification here:
+          <button
+            class="close-button"
+            @click=${e => e.target.dispatchEvent(new Event('close-overlay', { bubbles: true }))}
+          >
+            ⨯
+          </button>
+        </div>
+      </lion-demo-overlay>
+    `;
+  }}
+</Story>
+
+```js
+let shouldIntercept = true;
+let shouldInterceptButton;
+
+function toggleIntercept() {
+  shouldIntercept = !shouldIntercept;
+  shouldInterceptButton.textContent = shouldIntercept;
+}
+
+function intercept(ev) {
+  if (shouldIntercept) {
+    ev.preventDefault();
+  }
+}
+
+shouldInterceptButton = renderLitAsNode(
+  html`
+    <button @click="${toggleIntercept}">${shouldIntercept}</button>
+  `,
+);
+```
+
+```html
+<div>
+  Toggle shouldIntercept: ${shouldInterceptButton}
+</div>
+<lion-demo-overlay @before-closed=${intercept} @before-opened=${intercept}>
+  <button slot="invoker">Overlay</button>
+  <div slot="content" class="demo-overlay">
+    Hello! You can close this notification here:
+    <button
+      class="close-button"
+      @click=${e => e.target.dispatchEvent(new Event('close-overlay', { bubbles: true }))}
+    >
+      ⨯
+    </button>
+  </div>
+</lion-demo-overlay>
+```
+
+## Future
```

Comment:
Maybe put all of the below in a design considerations/ rationale document (Overlay System Scope).

Also:
- I think some of the props like `handlesAccessibility ` and `isTooltip ` are implemented?
- `isModal` is not needed since this has a 1:1 relation with hidesOnOutsideClick.
- `handlesUserInteraction` would be relevant for cases like these (multiple invokers): https://github.com/ing-bank/lion/pull/38 But we can omit for now